Mathlib/CategoryTheory/Limits/ConcreteCategory/WithAlgebraicStructures.lean

```diff
@@ -25,9 +26,37 @@ about `colimit F` when objects and morphisms in `C` have some algebraic structur
   and `x : Fᵢ` we have `r • x = 0` implies `x = 0`, then if `r • x = 0` for `x : colimit F`, then
   `x = 0`.
 
+Let `R` be the filtered colimit of rings `{Rⱼ}` and `M` be the filtered colimit of abelian groups
+`{Mⱼ}`  with `j` running through the same filtered index category `J`. If for each `j ∈ J`, `Mⱼ` is
+an `Rⱼ`-module such that the `Rⱼ`-actions are compatible, then `M` is an `Rⱼ`-module for all `j`
+and `M` is an `R`-module.
+
+- `Module.overFilteredColimits.moduleColimitColimit`: Let `R` and `M` be as above, then `M` is an
+  `R`-module.
+
+- `Module.overFilteredColimits.smul_spec`: Let `R` and `M` be as above, then the `R`-action on `M`
+  is given by the following: if `s : Rⱼ₁` and `t : Mⱼ₂` where `j₁ ≤ j₃` and `j₂ ≤ j₃`, then
+  `[s] • [m]` is equal to `[s|ⱼ₃ • m |ⱼ₃]` where `[...]` is the equivalence class in the colimit.
+
+## Implementation notes on `Module.overFilteredColimits`
+
+1. We choose not to use `PresheafOfModules` to avoid code duplication:
+consider `R : J ⥤ CommRingCat` and `M : J ⥤ AddCommGrp`, then `colimit M` is both a
+`colimit R`-module and a `colimit (R ⋙ forget₂ CommRingCat RingCat)`-module; the two module
+structures are virtually the same. This situation manifests in stalks of sheaves of modules:
+for any ringed space `X` and a sheaf of `𝒪_X`-module `ℳ`, we want to think the stalk `ℳₓ` as an
+`𝒪_{X,x}`-module. But since `PresheafOfModules` requires a presheaf of `RingCat` not `CommRingCat`,
+we need to compose the sheaf with forgetful functors, but we don't want to think about the
+difference between `𝒪_{X, x}` as a colimit in `CommRing` and `𝒪_{X, x}` as a colimit in `RingCat`
+all the time. So we ask `R` and `M` to be functors into concrete categories which behaves like rings
+and abelian groups respectively.
```

Comment:
In order to avoid this issue, it seems to me that you could use any colimit cocone in `RingCat` instead of `colimit R`.

Reply:
Sorry for the late reply, I completely forgot about this PR! Can you clarify the idea please? Basically, for a scheme $X$ and $O_X$-module $M$, I want to define module structure of $M_x$ over `TopCat.Presheaf.stalk X.presheaf x` and `TopCat.Presheaf.stalk X.ringCatSheaf.1 x` but I only want to define it once.

Reply:
In the comment I am referring to, you seem to have an issue with applying `colimit` in the category `RingCat` because you would like to apply `colimit` in `CommRingCat`. What I suggest is that instead of using `colimit`, you could develop your API using any colimit cocone in `RingCat`.